core/json-utils/src/main/java/software/amazon/awssdk/protocols/jsoncore/JsonWriter.java

```diff
@@ -0,0 +1,288 @@
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(String val) {
+        try {
+            generator.writeString(val);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(boolean bool) {
+        try {
+            generator.writeBoolean(bool);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(long val) {
+        try {
+            generator.writeNumber(val);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(double val) {
+        try {
+            generator.writeNumber(val);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(float val) {
+        try {
+            generator.writeNumber(val);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(short val) {
+        try {
+            generator.writeNumber(val);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(int val) {
+        try {
+            generator.writeNumber(val);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(ByteBuffer bytes) {
+        try {
+            generator.writeBinary(BinaryUtils.copyBytesFrom(bytes));
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(Instant instant) {
+        try {
+            generator.writeNumber(DateUtils.formatUnixTimestampInstant(instant));
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(BigDecimal value) {
+        try {
+            generator.writeString(value.toString());
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(BigInteger value) {
+        try {
+            generator.writeNumber(value);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeNumber(String number) {
+        try {
+            generator.writeNumber(number);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
```

Comment:
minor: this try-catch repeats throughout, would it help readability to have a wrapper around it?

```java
unsafeWrite(() -> generator.writeNumber(number));
```

Reply:
I like it!